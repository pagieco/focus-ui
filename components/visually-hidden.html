<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Visually hidden | Focus UI Docs</title>
    <meta name="description" content="">
    
    
    <link rel="preload" href="/focus-ui/assets/css/0.styles.7c918269.css" as="style"><link rel="preload" href="/focus-ui/assets/js/app.ec7e616f.js" as="script"><link rel="preload" href="/focus-ui/assets/js/2.f856ce09.js" as="script"><link rel="preload" href="/focus-ui/assets/js/45.533789bc.js" as="script"><link rel="prefetch" href="/focus-ui/assets/js/10.89c71ea6.js"><link rel="prefetch" href="/focus-ui/assets/js/11.51ec99ee.js"><link rel="prefetch" href="/focus-ui/assets/js/12.696b690b.js"><link rel="prefetch" href="/focus-ui/assets/js/13.e6a8f5e3.js"><link rel="prefetch" href="/focus-ui/assets/js/14.99473aba.js"><link rel="prefetch" href="/focus-ui/assets/js/15.33f3f7ef.js"><link rel="prefetch" href="/focus-ui/assets/js/16.e175e0d8.js"><link rel="prefetch" href="/focus-ui/assets/js/17.92fb1298.js"><link rel="prefetch" href="/focus-ui/assets/js/18.0edd0496.js"><link rel="prefetch" href="/focus-ui/assets/js/19.2ed41cc3.js"><link rel="prefetch" href="/focus-ui/assets/js/20.bbc97549.js"><link rel="prefetch" href="/focus-ui/assets/js/21.e60a114a.js"><link rel="prefetch" href="/focus-ui/assets/js/22.d3bbd8fc.js"><link rel="prefetch" href="/focus-ui/assets/js/23.ce398370.js"><link rel="prefetch" href="/focus-ui/assets/js/24.ed77a66d.js"><link rel="prefetch" href="/focus-ui/assets/js/25.4fdc0e63.js"><link rel="prefetch" href="/focus-ui/assets/js/26.08cb5c31.js"><link rel="prefetch" href="/focus-ui/assets/js/27.5414d0b8.js"><link rel="prefetch" href="/focus-ui/assets/js/28.eb3beb40.js"><link rel="prefetch" href="/focus-ui/assets/js/29.914015b8.js"><link rel="prefetch" href="/focus-ui/assets/js/3.40754a46.js"><link rel="prefetch" href="/focus-ui/assets/js/30.fb1c4726.js"><link rel="prefetch" href="/focus-ui/assets/js/31.978a5344.js"><link rel="prefetch" href="/focus-ui/assets/js/32.6c9df44b.js"><link rel="prefetch" href="/focus-ui/assets/js/33.4f3ddca0.js"><link rel="prefetch" href="/focus-ui/assets/js/34.e1d7f155.js"><link rel="prefetch" href="/focus-ui/assets/js/35.db326522.js"><link rel="prefetch" href="/focus-ui/assets/js/36.4470a8fe.js"><link rel="prefetch" href="/focus-ui/assets/js/37.e3d7fa2b.js"><link rel="prefetch" href="/focus-ui/assets/js/38.fc133bf8.js"><link rel="prefetch" href="/focus-ui/assets/js/39.6ec60546.js"><link rel="prefetch" href="/focus-ui/assets/js/4.11d613bd.js"><link rel="prefetch" href="/focus-ui/assets/js/40.7967dec1.js"><link rel="prefetch" href="/focus-ui/assets/js/41.e4d9d7b4.js"><link rel="prefetch" href="/focus-ui/assets/js/42.bcb427ce.js"><link rel="prefetch" href="/focus-ui/assets/js/43.9c91ae94.js"><link rel="prefetch" href="/focus-ui/assets/js/44.50e43115.js"><link rel="prefetch" href="/focus-ui/assets/js/46.116d38c9.js"><link rel="prefetch" href="/focus-ui/assets/js/5.2718a8ea.js"><link rel="prefetch" href="/focus-ui/assets/js/6.bb50d40e.js"><link rel="prefetch" href="/focus-ui/assets/js/7.dee98ecb.js"><link rel="prefetch" href="/focus-ui/assets/js/8.374f9f82.js"><link rel="prefetch" href="/focus-ui/assets/js/9.4b4b3660.js">
    <link rel="stylesheet" href="/focus-ui/assets/css/0.styles.7c918269.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/focus-ui/" class="home-link router-link-active"><!----> <span class="site-name">Focus UI Docs</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <!----></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><!---->  <ul class="sidebar-links"><li><a href="/focus-ui/" class="sidebar-link">Introduction</a></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading open"><span>Components</span> <span class="arrow down"></span></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/focus-ui/components/avatar.html" class="sidebar-link">Avatar</a></li><li><a href="/focus-ui/components/badge.html" class="sidebar-link">Badge</a></li><li><a href="/focus-ui/components/banner.html" class="sidebar-link">Banner</a></li><li><a href="/focus-ui/components/button.html" class="sidebar-link">Button</a></li><li><a href="/focus-ui/components/button-group.html" class="sidebar-link">Button group</a></li><li><a href="/focus-ui/components/card.html" class="sidebar-link">Card</a></li><li><a href="/focus-ui/components/checkbox.html" class="sidebar-link">Checkbox</a></li><li><a href="/focus-ui/components/color-picker.html" class="sidebar-link">Color Picker</a></li><li><a href="/focus-ui/components/data-table.html" class="sidebar-link">Data table</a></li><li><a href="/focus-ui/components/display-text.html" class="sidebar-link">Display text</a></li><li><a href="/focus-ui/components/footer-help.html" class="sidebar-link">Footer help</a></li><li><a href="/focus-ui/components/form.html" class="sidebar-link">Form</a></li><li><a href="/focus-ui/components/form-layout.html" class="sidebar-link">Form layout</a></li><li><a href="/focus-ui/components/frame.html" class="sidebar-link">Frame</a></li><li><a href="/focus-ui/components/heading.html" class="sidebar-link">Heading</a></li><li><a href="/focus-ui/components/icon.html" class="sidebar-link">Icon</a></li><li><a href="/focus-ui/components/layout.html" class="sidebar-link">Layout</a></li><li><a href="/focus-ui/components/link.html" class="sidebar-link">Link</a></li><li><a href="/focus-ui/components/list.html" class="sidebar-link">List</a></li><li><a href="/focus-ui/components/modal.html" class="sidebar-link">Modal</a></li><li><a href="/focus-ui/components/option-list.html" class="sidebar-link">Option list</a></li><li><a href="/focus-ui/components/page.html" class="sidebar-link">Page</a></li><li><a href="/focus-ui/components/pagination.html" class="sidebar-link">Pagination</a></li><li><a href="/focus-ui/components/popover.html" class="sidebar-link">Popover</a></li><li><a href="/focus-ui/components/resource-list.html" class="sidebar-link">Resource list</a></li><li><a href="/focus-ui/components/select.html" class="sidebar-link">Select</a></li><li><a href="/focus-ui/components/skeleton-body-text.html" class="sidebar-link">Skeleton body text</a></li><li><a href="/focus-ui/components/skeleton-display-text.html" class="sidebar-link">Skeleton display text</a></li><li><a href="/focus-ui/components/skeleton-page.html" class="sidebar-link">Skeleton page</a></li><li><a href="/focus-ui/components/skeleton-thumbnail.html" class="sidebar-link">Skeleton thumbnail</a></li><li><a href="/focus-ui/components/spinner.html" class="sidebar-link">Spinner</a></li><li><a href="/focus-ui/components/stack.html" class="sidebar-link">Stack</a></li><li><a href="/focus-ui/components/sub-heading.html" class="sidebar-link">Subheading</a></li><li><a href="/focus-ui/components/text-container.html" class="sidebar-link">Text container</a></li><li><a href="/focus-ui/components/text-field.html" class="sidebar-link">Text Field</a></li><li><a href="/focus-ui/components/text-style.html" class="sidebar-link">Text style</a></li><li><a href="/focus-ui/components/thumbnail.html" class="sidebar-link">Thumbnail</a></li><li><a href="/focus-ui/components/toast.html" class="sidebar-link">Toast</a></li><li><a href="/focus-ui/components/tooltip.html" class="sidebar-link">Tooltip</a></li><li><a href="/focus-ui/components/visually-hidden.html" class="active sidebar-link">Visually hidden</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/focus-ui/components/visually-hidden.html#best-practices" class="sidebar-link">Best practices</a></li><li class="sidebar-sub-header"><a href="/focus-ui/components/visually-hidden.html#content-guidelines" class="sidebar-link">Content guidelines</a></li><li class="sidebar-sub-header"><a href="/focus-ui/components/visually-hidden.html#examples" class="sidebar-link">Examples</a></li><li class="sidebar-sub-header"><a href="/focus-ui/components/visually-hidden.html#accessibility" class="sidebar-link">Accessibility</a></li></ul></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="visually-hidden"><a href="#visually-hidden" aria-hidden="true" class="header-anchor">#</a> Visually hidden</h1> <p>Use when an element needs to be available to assistive technology (for example, a screen reader) but otherwise hidden.</p> <hr> <h2 id="best-practices"><a href="#best-practices" aria-hidden="true" class="header-anchor">#</a> Best practices</h2> <p>Visually hidden should:</p> <ul><li>Not be used if semantic markup can make content understandable to people using assistive technology</li> <li>Be used to provide extra context when semantic markup isn't enough</li> <li>Be used on any content that is normally present but is being omitted</li> <li>Make sense in context when used with a screen reader</li></ul> <hr> <h2 id="content-guidelines"><a href="#content-guidelines" aria-hidden="true" class="header-anchor">#</a> Content guidelines</h2> <p>There are no content elements specific for this component. Follow any guidelines appropriate to the element being
hidden.</p> <hr> <h2 id="examples"><a href="#examples" aria-hidden="true" class="header-anchor">#</a> Examples</h2> <h3 id="visually-hidden-heading"><a href="#visually-hidden-heading" aria-hidden="true" class="header-anchor">#</a> Visually hidden heading</h3> <p>Always provide a heading for a major page section such as a card. In rare cases the heading is visually redundant and
the meaning is conveyed by context. Rather than omit the heading, wrap the heading in the visually hidden component.</p> <div class="language-vue extra-class"><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Card</span> <span class="token attr-name">sectioned</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>VisuallHidden</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Heading</span><span class="token punctuation">&gt;</span></span>Title and description<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Heading</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>VisuallHidden</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>FormLayout</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>TextField</span>
      <span class="token attr-name">label</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">&quot;</span>Title<span class="token punctuation">&quot;</span></span>
      <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">&quot;</span>Homepage<span class="token punctuation">&quot;</span></span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>FormLayout</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Card</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><hr> <h2 id="accessibility"><a href="#accessibility" aria-hidden="true" class="header-anchor">#</a> Accessibility</h2> <p>The visually hidden component styles text so that it's not visible, but it is available to assistive technologies like
screen readers and other text to speech programs.</p> <p>The component shouldn't be used to hide interactive content.</p></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
        ←
        <a href="/focus-ui/components/tooltip.html" class="prev">
          Tooltip
        </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/focus-ui/assets/js/app.ec7e616f.js" defer></script><script src="/focus-ui/assets/js/2.f856ce09.js" defer></script><script src="/focus-ui/assets/js/45.533789bc.js" defer></script>
  </body>
</html>
