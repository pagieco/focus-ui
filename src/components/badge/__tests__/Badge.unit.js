import { mount, shallowMount } from '@vue/test-utils';
import {
  default as Badge,
  Size as BadgeSize,
  Status as BadgeStatus,
} from '../index';
import { VisuallyHidden } from '../../visually-hidden';

const shallowWrap = (propsData = {}) => shallowMount(Badge, { propsData });
const wrap = (propsData = {}) => mount(Badge, { propsData });

describe('Badge', () => {
  it('is a valid vue component', () => {
    expect(Badge).toBeAComponent();
  });

  it('matches the snapshot', () => {
    expect(shallowMount(Badge).element).toMatchSnapshot();
  });

  it('uses the size prop', () => {
    const wrapper = shallowWrap({ size: BadgeSize.Small });

    expect(wrapper.classes('badge--size-small')).toBeTruthy();
  });

  it('validates the size prop', () => {
    const prop = Badge.props.size;

    expect(prop.validator).toBeInstanceOf(Function);
    expect(prop.validator('small')).toBeTruthy();
    expect(prop.validator('medium')).toBeTruthy();
    expect(prop.validator('huge')).toBeFalsy();
  });

  it('uses the status prop', () => {
    const wrapper = shallowWrap({ status: BadgeStatus.Success });

    expect(wrapper.classes('badge--status-success')).toBeTruthy();
  });

  it('validates the status prop', () => {
    const prop = Badge.props.status;

    expect(prop.validator).toBeInstanceOf(Function);
    expect(prop.validator('success')).toBeTruthy();
    expect(prop.validator('info')).toBeTruthy();
    expect(prop.validator('attention')).toBeTruthy();
    expect(prop.validator('danger')).toBeFalsy();
  });

  it('renders the visually-hidden component when the status prop is given', () => {
    const wrapper = wrap({ status: BadgeStatus.Success });

    expect(wrapper.find(VisuallyHidden).exists()).toBeTruthy();
  });
});
