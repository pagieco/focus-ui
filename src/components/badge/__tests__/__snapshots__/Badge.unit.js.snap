// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge matches the snapshot 1`] = `
<span
  class="badge badge--size-medium"
>
  <!---->
   
</span>
`;
